src/slack-send.js

```diff
@@ -0,0 +1,95 @@
+const github = require('@actions/github');
+const { WebClient } = require('@slack/web-api');
+const flatten = require('flat');
+const axios = require('axios');
+
+const SLACK_WEBHOOK_TYPES = {
+  WORKFLOW_TRIGGER: 'WORKFLOW_TRIGGER',
+  INCOMING_WEBHOOK: 'INCOMING_WEBHOOK',
+};
+
+module.exports = async function slackSend(core) {
+  try {
+    const botToken = process.env.SLACK_BOT_TOKEN;
+    const webhookUrl = process.env.SLACK_WEBHOOK_URL;
+    let webhookType = SLACK_WEBHOOK_TYPES.WORKFLOW_TRIGGER;
+
+    if (process.env.SLACK_WEBHOOK_TYPE) {
+      // The default type is for Workflow Builder triggers. If you want to use this action for Incoming Webhooks, use
+      // the corresponding type instead.
+      webhookType = process.env.SLACK_WEBHOOK_TYPE.toUpperCase();
+    }
+
+    if (botToken === undefined && webhookUrl === undefined) {
+      throw new Error('Need to provide at least one botToken or webhookUrl');
+    }
+
+    let payload = core.getInput('payload');
+
+    if (payload) {
+      try {
+        // confirm it is valid json
+        payload = JSON.parse(payload);
+      } catch (e) {
+        // passed in payload wasn't valid json
+        console.error('passed in payload was invalid JSON');
```

Comment:
I think it's fine to use `console` for this purpose. Should we ignore "no-console" warnings for this entire package?

```
> slack-github-action@1.15.0 lint
> eslint .


/Users/ksera/github/slack-github-action/src/slack-send.js
  35:9  warning  Unexpected console statement  no-console
  49:9  warning  Unexpected console statement  no-console
  56:9  warning  Unexpected console statement  no-console
  77:9  warning  Unexpected console statement  no-console
  78:9  warning  Unexpected console statement  no-console

✖ 5 problems (0 errors, 5 warnings)
```

Reply:
Yes that makes sense. I'll add it to the eslint rules for this repo.